package/MDAnalysis/core/groups.py

```diff
@@ -509,29 +511,30 @@ def rotate(self, R):
         .. math::
 
             \mathbf{x}' = \mathsf{R}\mathbf{x}
-        """
-        atomgroup = self.atoms
 
-        R = np.matrix(R, copy=False, dtype=np.float32)
+        """
+        R = np.asarray(R)
         # changes the coordinates (in place)
-        x = atomgroup.universe.trajectory.ts.positions
-        idx = atomgroup.indices
-        x[idx] = x[idx] * R.T  # R.T acts to the left & is broadcasted N times.
+        x = self.atoms.unique.universe.trajectory.ts.positions
```

Comment:
all the other functions that I've tested here also use `unique` so I just followed their style.